ts/input/tex/textmacros/TextMacrosMethods.ts

```diff
@@ -0,0 +1,284 @@
+   */
+  Tilde(parser: TextParser, _c: string) {
+    parser.text += '\u00A0';  // non-breaking space
+  },
+
+  /**
+   * @param {TextParser} parser   The text-mode parser
+   * @param {string} c            The character that called this function
+   */
+  Space(parser: TextParser, _c: string) {
+    parser.text += ' ';  // regular space, but skipping multiple spaces
+    while (parser.GetNext().match(/\s/)) parser.i++;
+  },
+
+  /**
+   * @param {TextParser} parser   The text-mode parser
+   * @param {string} name         The control sequence that called this function
+   */
+  SelfQuote(parser: TextParser, name: string) {
+    parser.text += name.substr(1);  // add in the quoted character
+  },
+
+  /**
+   * @param {TextParser} parser   The text-mode parser
+   * @param {string} name         The control sequence that called this function
+   * @param {string} c            The character to insert into the string
+   */
+  Insert(parser: TextParser, _name: string, c: string) {
+    parser.text += c;
+  },
+
+  /**
+   * @param {TextParser} parser   The text-mode parser
+   * @param {string} name         The control sequence that called this function
+   * @param {string} c            The character to insert into the string
+   */
+  Accent(parser: TextParser, _name: string, c: string) {
+    //
+    //  Create an accented character using mover
+    //
+    const base = parser.ParseArg(name);
+    const accent = parser.create('token', 'mo', {}, c);
+    parser.addAttributes(accent);
+    parser.Push(parser.create('node', 'mover', [base, accent]));
+  },
+
+  /**
+   * @param {TextParser} parser   The text-mode parser
+   * @param {string} name         The control sequence that called this function
+   */
+  Emph(parser: TextParser, name: string) {
+    //
+    //  Switch to/from italics
+    //
+    const variant = (parser.stack.env.mathvariant === '-tex-mathit' ? 'normal' : '-tex-mathit');
+    parser.Push(parser.ParseTextArg(name, {mathvariant: variant}));
+  },
+
+  /**
+   * @param {TextParser} parser   The text-mode parser
+   * @param {string} name         The control sequence that called this function
+   * @param {string} variant      The font variant to use from now on
+   */
+  SetFont(parser: TextParser, _name: string, variant: string) {
+    parser.saveText();
+    parser.stack.env.mathvariant = variant;
+  },
+
+  /**
+   * @param {TextParser} parser   The text-mode parser
+   * @param {string} name         The control sequence that called this function
+   * @param {number} size         The font size to use from now on
+   */
+  SetSize(parser: TextParser, _name: string, size: number) {
+    parser.saveText();
+    parser.stack.env.mathsize = size;
+  },
+
+  /**
+   * @param {TextParser} parser   The text-mode parser
+   * @param {string} name         The control sequence that called this function
+   */
+  CheckAutoload(parser: TextParser, name: string) {
+    const autoload = parser.configuration.packageData.autoload;
```

Comment:
Could those macros not be included into the `Autoload` package? Or is that meant to work even if `Autoload` is not improrted?

Reply:
This function checks if a macro is set up to be autoloaded, which will only be true if the `autoload` package has been included in the TeX parser, and if the macro is included in the configuration for `autoload`.  

When it is set to be autoloaded, we need to perform that autoloading in the original TeX parser, not the parser that is handling the text macros.  That is why we check for it being an autoloaded macro.  (The original TeX parser is also used when the macro is not defined so that the proper error message or fallback action is performed.)  If it was autoloaded, the expression is restarted, so that the proper macro definition will be used.  Otherwise, the text parser is used (line 267 uses `[texParser, name]` while line 271 uses `[parser, name]`).

The upshot is, this will still work if `autoload` extension is not loaded, or not included in the packages for the TeX parser, or if the macro is not configured to autoload.

Reply:
I've improved the comments to try to make this clearer.  But I believe the code is correct as is.

Reply:
I think the new comment clarifies my confusion: If `autoload` is there it will have filled the `packageData` and we ask `TexParser` to handle it. Thanks.

Reply:
Glad that helped out.  It is a bit complicated how that works, and it took me a bit to remember the details.